tests/posix/common/src/clock.c

```diff
@@ -70,8 +71,31 @@ void test_posix_realtime(void)
 	ret = clock_gettime(CLOCK_REALTIME, &rts);
 	zassert_equal(ret, 0, "Fail to get realtime clock");
 
-	zassert_equal(rts.tv_sec, mts.tv_sec, "Seconds not equal");
-	zassert_equal(rts.tv_nsec, mts.tv_nsec, "Nanoseconds not equal");
+	/* Between both clock_gettime(), time may have been increased by 1ms
+	 * (precision of k_uptime_get()). This could imply a rollover of nsec
+	 * counter and increment number of seconds.
+	 */
+	if ((rts.tv_nsec != mts.tv_nsec) || (rts.tv_sec != mts.tv_sec)) {
+		/* If nsec or sec are not equal, then rts should be equal to
+		 * mts + 1ms
+		 */
+		mts_tvnsec_plus1 = mts.tv_nsec +
+			(USEC_PER_MSEC * NSEC_PER_USEC);
+
+		if (mts_tvnsec_plus1 >= NSEC_PER_SEC) {
+			/* nsec counter rollover */
+			mts_tvnsec_plus1 -= NSEC_PER_SEC;
+			zassert_equal(rts.tv_sec, mts.tv_sec + 1,
+				"Seconds neither equal, nor differ by 1");
+		} else {
+			zassert_equal(rts.tv_sec, mts.tv_sec,
+				"Seconds neither equal, nor differ by 1");
+		}
+		zassert_equal(rts.tv_nsec, mts_tvnsec_plus1,
+			"Nanoseconds neither equal, nor differ by 1");
+	} else {
```

Comment:
I'd suggest to remove this empty else. It doesn't add clarity, just make an already big `if` even more unwieldy.